homeassistant/components/samsungtv/config_flow.py

```diff
@@ -74,38 +70,24 @@ def _get_entry(self):
                 CONF_ID: self._id,
                 CONF_IP_ADDRESS: self._ip,
                 CONF_MANUFACTURER: self._manufacturer,
-                CONF_METHOD: self._method,
+                CONF_METHOD: self._bridge.method,
                 CONF_MODEL: self._model,
                 CONF_NAME: self._name,
-                CONF_PORT: self._port,
+                CONF_PORT: self._bridge.port,
+                CONF_TOKEN_FILE: self._bridge.token_file,
             },
         )
 
     def _try_connect(self):
         """Try to connect and check auth."""
-        for cfg in SUPPORTED_METHODS:
-            config = {
-                "name": "HomeAssistant",
-                "description": "HomeAssistant",
-                "id": "ha.component.samsung",
-                "host": self._host,
-                "port": self._port,
-            }
-            config.update(cfg)
-            try:
-                LOGGER.debug("Try config: %s", config)
-                with Remote(config.copy()):
-                    LOGGER.debug("Working config: %s", config)
-                    self._method = cfg["method"]
-                    return RESULT_SUCCESS
-            except AccessDenied:
-                LOGGER.debug("Working but denied config: %s", config)
-                return RESULT_AUTH_MISSING
-            except (UnhandledResponse, WebSocketException):
-                LOGGER.debug("Working but unsupported config: %s", config)
-                return RESULT_NOT_SUPPORTED
-            except OSError as err:
-                LOGGER.debug("Failing config: %s, error: %s", config, err)
+        self._bridge = SamsungTVWSBridge(self.hass)
+        result = self._bridge.try_connect(self._host, self._port)
+        if result == RESULT_SUCCESS:
+            return result
+        self._bridge = SamsungTVLegacyBridge()
+        result = self._bridge.try_connect(self._host, self._port)
+        if result == RESULT_SUCCESS:
+            return result
```

Comment:
You should use `get_bridge` to abstract the child classes completely.

Reply:
Sounds good